index.d.ts

```diff
@@ -0,0 +1,38 @@
+type TimeWindow = number; // e.g. 60 * 1000
+type Snapshot = object;
```

Comment:
`Snapshot` type needs to have methods, `reset()` and `add(snapshot: Snapshot)`. Could you put these methods?